0_download_data/correctionutils.py

```diff
@@ -0,0 +1,114 @@
+"""
+This file contains functions to perform the splitting, cropping, and conversion of the second NF1 dataset using CellProfiler and
+to reorder and rename the metadata of the images to fit the standard from the pilot dataset.
+"""
+import os
+import pathlib
+from pathlib import Path
+
+
+def correct_images(path_to_pipeline: str, path_to_output: str, path_to_images: str):
+    """
+    correct NF1 second plate with CellProfiler
+
+    Parameters
+    ----------
+        path_to_pipeline (str): path to the CellProfiler .cppipe file
+        path_to_output (str): path to the output folder for the corrected images
+        path_to_images (str): path to folder with raw second plate images
+    """
+    if len(os.listdir(path_to_output)) == 0:
+        # run CellProfiler on a plate that has not been analyzed yet (e.g. no files)
+        command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
+        os.system(command)
+    else:
+        print("This plate has already been ran through CellProfiler!")
+
+
+def rename_images(path_to_images: pathlib.Path, output_folder_name: str):
+    """
+    reorder and rename all images
+
+    Parameters
+    ----------
+    path_to_images : pathlib.Path
+        path to where the corrected images from CellProfiler are located
+    output_folder_name:
+        name of the folder with the outputed
+    """
+    original_prefix = ("DAPI", "GFP", "Actin")
+
+    if any(
+        images.name.startswith(original_prefix) for images in path_to_images.iterdir()
+    ):
+        for images in path_to_images.iterdir():
+            image_names = images.name.split("_")
+            well = image_names[1]
+            site = image_names[2]
+            channel_name = image_names[0]
+            plate = image_names[3].split(".")[0]
+            new_image_name = (
+                f"{output_folder_name}/{well}_01_{site}_{channel_name}_{plate}.tif"
+            )
+            Path(images).rename(Path(new_image_name))
+            print("The image names are reordered")
+    else:
+        print("This plate has already been reordered!")
+
+    dapi_id = "DAPI"
+    gfp_id = "GFP"
+    actin_id = "Actin"
+
+    keywords = (
+        "01_1_1",
+        "01_1_2",
+        "01_1_3",
+        "01_1_4",
+        "01_2_1",
+        "01_2_2",
+        "01_2_3",
+        "01_2_4",
+        "01_3_1",
+        "01_3_2",
+        "01_3_3",
+        "01_3_4",
+    )
+
+    for images in path_to_images.iterdir():
+        if any(x in str(images) for x in keywords):
+            print("The metadata for this plate has already been corrected!")
+            break
+        else:
```

Comment:
Consider removing this `else` and unindenting the if blocks below. The `break` in your if-case above ensures the process wouldn't run anything below were it triggered.
```suggestion
```

Reply:
Oooh I didn't even think of that. Fixed!